### ISSUE TITLE
resource `meraki_networks_wireless_settings` settings - error on upgrade strategy

### ISSUE THREAD

**Issue by finkjordanj** [cisco-open/terraform-provider-meraki#54]
**Prerequisites**
* [x] Have you tested the operation in the API directly?
* [x] Do you have the latest Terraform provider version?
* [x] Review the [compatibility matrix](https://github.com/meraki/terraform-provider-meraki/tree/main#compatibility-matrix) before opening an issue. 

**Describe the bug**
The upgrade strategy appears to not be setup with the same input as the API is expecting.
When applying the following resource get this error on apply.

```
resource "meraki_networks_wireless_settings" "this" {
  network_id                 = local.this_network.id
  ipv6_bridge_enabled        = true
  led_lights_on              = true
  location_analytics_enabled = true
  meshing_enabled            = true
  upgradestrategy            = "minimizeUpgradeTime"
}
```
Appears to be issue with how this variable is called out at the function call.
https://github.com/cisco-open/terraform-provider-meraki/blob/9b712fa00fbd4b6978f8e3864b7fa64368994e8c/internal/provider/resource_meraki_networks_wireless_settings.go#L145C5-L145C20
Input is listed as upgradestrategy, but format is setup via the tfsdk as upgrade_strategy. 

**Expected behavior**
Apply successfully but not currently.

**Screenshots**
Error on current apply.
```
fink@LTWXCGVT333:~/dev/meraki-terraform-test $ terraform apply
meraki_devices.this_ms_sw1: Refreshing state... [name=750-comm-meraki-lab-sw1]
meraki_devices.this_ms_ext_sw1: Refreshing state... [name=750-comm-meraki-lab-ext-sw1]
meraki_devices.this_ms_sw2: Refreshing state... [name=750-comm-meraki-lab-sw2]
meraki_devices.this_mx: Refreshing state... [name=750-comm-meraki-lab-mx1]
data.meraki_networks.dev_networks: Reading...
data.meraki_networks.dev_networks: Read complete after 1s
meraki_networks_switch_settings.this_site_switch_settings: Refreshing state...
meraki_networks_appliance_firewall_firewalled_services.this_web: Refreshing state...
meraki_networks_appliance_vlans_settings.this_vlans: Refreshing state...
meraki_networks_switch_mtu.this_site_switch_mtu: Refreshing state...
meraki_networks_appliance_vlans.this_vlan_1: Refreshing state... [id=1]
meraki_networks_syslog_servers.this_spoke: Refreshing state...
meraki_networks_appliance_firewall_firewalled_services.this_snmp: Refreshing state...
meraki_networks_appliance_settings.this: Refreshing state...
meraki_networks_group_policies.group_policy_byod: Refreshing state... [name=BYOD]
meraki_networks_appliance_firewall_firewalled_services.this_icmp: Refreshing state...
meraki_networks_appliance_vlans.this_vlan_20: Refreshing state... [id=20]
meraki_networks_appliance_vlans.this_vlan_5: Refreshing state... [id=5]
meraki_networks_appliance_vlans.this_vlan_30: Refreshing state... [id=30]
meraki_networks_appliance_vlans.this_vlan_10: Refreshing state... [id=10]
meraki_networks_appliance_vlans.this_vlan_2: Refreshing state... [id=2]
meraki_networks_group_policies.group_policy_guest: Refreshing state... [name=GUEST]
meraki_networks_group_policies.group_policy_pan: Refreshing state... [name=PAN]
meraki_networks_group_policies.group_policy_internal: Refreshing state... [name=INTERNAL]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create
  ~ update in-place

Terraform will perform the following actions:

  # meraki_networks_wireless_settings.this will be created
  + resource "meraki_networks_wireless_settings" "this" {
      + ipv6_bridge_enabled        = true
      + led_lights_on              = true
      + location_analytics_enabled = true
      + meshing_enabled            = true
      + named_vlans                = (known after apply)
      + network_id                 = "L_783626335162467749"
      + regulatory_domain          = (known after apply)
      + upgradestrategy            = "minimizeUpgradeTime"
    }

Plan: 1 to add, 4 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

meraki_networks_wireless_settings.this: Creating...
meraki_devices.this_ms_ext_sw1: Modifying... [name=750-comm-meraki-lab-ext-sw1]
meraki_devices.this_ms_sw1: Modifying... [name=750-comm-meraki-lab-sw1]
meraki_devices.this_mx: Modifying... [name=750-comm-meraki-lab-mx1]
meraki_devices.this_ms_sw2: Modifying... [name=750-comm-meraki-lab-sw2]
meraki_devices.this_mx: Modifications complete after 1s [name=750-comm-meraki-lab-mx1]
meraki_devices.this_ms_ext_sw1: Modifications complete after 1s [name=750-comm-meraki-lab-ext-sw1]
meraki_devices.this_ms_sw2: Modifications complete after 1s [name=750-comm-meraki-lab-sw2]
meraki_devices.this_ms_sw1: Modifications complete after 1s [name=750-comm-meraki-lab-sw1]
╷
│ Error: Value Conversion Error
│ 
│   with meraki_networks_wireless_settings.this,
│ An unexpected error was encountered trying to convert tftypes.Value into provider.NetworksWirelessSettingsRs. This is always an error in the provider. Please report the following to the provider developer:
│ 
│ mismatch between struct and object: Struct defines fields not found in object: upgrade_strategy. Object defines fields not found in struct: upgradestrategy.
```

**Environment (please complete the following information):**
* Meraki Dashboard version: 
* Terraform version:1.8.3
* Meraki provider version: 0.2.1-alpha
* OS Version: 22.04 Ubuntu

**Additional context**
Removed state refresh on meraki_device hardware.

**Comment by fmunozmiranda** [comment]
Hi @finkjordanj could you please try it again with v0.2.2-apha and update if it works now?

**Comment by finkjordanj** [comment]
This appears to be resolved in 0.2.2-alpha. Apply and state refresh worked correctly now.
